test/Mvc.Analyzers.Test/TestFiles/TopLevelParameterAttributeAnalyzerTest/DiagnosticsAreReturned_ForControllerActionsWithParametersWithRequiredAttributeForIEnumerable.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers.TestFiles.TopLevelParameterAttributeAnalyzerTest
+{
+   public class DiagnosticsAreReturned_ForControllerActionsWithParametersWithRequiredAttributeForIEnumerable : Controller
+    {
+        public IActionResult Index([Required]IEnumerable<int> /*MM*/collection) => null;
```

Comment:
`IEnumerable<T>` does not inherit from `ICollection`. So, I suspect the analyzer would also complain about a `[Required] int nonCollection` parameter.

Reply:
Yes, you're right @dougbu  will fix this.